src/lib.rs

```diff
@@ -447,15 +447,29 @@ impl Options {
         line
     }
 
-    /// Derive a usage message from a set of options.
+
+    /// Derive a formatted message from a set of options.
     pub fn usage(&self, brief: &str) -> String {
+        self.usage_with_format(|opts|
+            format!("{}\n\nOptions:\n{}\n", brief, opts.collect::<Vec<String>>().join("\n")))
+    }
+
+    /// Derive a custom formatted message from a set of options. The formatted options provided to
+    /// a closure as an iterator.
+    pub fn usage_with_format<F: FnMut(&mut Iterator<Item=String>) -> String>(&self, mut formatter: F) -> String {
+        let mut items = self.usage_items();
+        formatter(items.as_mut())
```

Comment:
Not that iot matters that much, but this should be writable as `formatter(&mut self.usage_items())` I think.

Reply:
Ah, yes, my bad. Artifacts of trying things after the discussion in IRC -- forgot to simplify. Done.